cmd/kube-controller-manager/app/core.go

```diff
@@ -178,7 +178,7 @@ func startGarbageCollectorController(ctx ControllerContext) (bool, error) {
 	if err != nil {
 		return true, fmt.Errorf("failed to get supported resources from server: %v", err)
 	}
-	deletableResources := discovery.FilteredBy(discovery.SupportsAllVerbs{Verbs: []string{"delete"}}, preferredResources)
+	deletableResources := discovery.FilteredBy(discovery.SupportsAllVerbs{Verbs: []string{"list", "watch", "delete"}}, preferredResources)
```

Comment:
I just looked and we also use "update" to remove the finalizer (there is a todo to switch to patch when strategic merge supports removing an entry from a slice of a base type). I'll make the appropriate changes.